sdk/appservice/mgmt/src/main/java/com/azure/management/appservice/implementation/FunctionAppImpl.java

```diff
@@ -111,16 +116,20 @@ private void initializeFunctionService() {
             } catch (MalformedURLException e) {
                 throw logger.logExceptionAsError(new IllegalStateException(e));
             }
-            RestClient client =
-                manager()
-                    .restClient()
-                    .newBuilder()
-                    .withBaseUrl(baseUrl)
-                    .withPolicy(new FunctionAuthenticationPolicy(this))
-                    .buildClient();
-            functionServiceHost = client.getBaseUrl().toString();
+
+            List<HttpPipelinePolicy> policies = new ArrayList<>();
+            for (int i = 0, count = manager().httpPipeline().getPolicyCount(); i < count; ++i) {
+                policies.add(manager().httpPipeline().getPolicy(i));
+            }
+            policies.add(new FunctionAuthenticationPolicy(this));
+            HttpPipeline httpPipeline= new HttpPipelineBuilder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(manager().httpPipeline().getHttpClient())
+                .build();
+            //TODO: confirm host url
+            functionServiceHost = manager().environment().getResourceManagerEndpoint();
```

Comment:
This is not correct. Should be the `baseUrl` above.

Reply:
For my understanding, it will be resource manager endpoint too as per the flow we start from `TestBase` to `AppServiceManager` to `FunctionAppImpl`. It seems there isn't any code snippet to change baseUrl after we set resouce manager endpoint initially.

Reply:
I wrote this so I am very sure. It is not mgmt client. It is like kudu client, host of which is `xxx.azurewebsite.net`